avro-builder/builder/src/main/java/com/linkedin/avroutil1/builder/operations/codegen/own/AvroUtilCodeGenOp.java

```diff
@@ -91,10 +93,53 @@ public void run(OperationContext opContext) throws Exception {
     long parseEnd = System.currentTimeMillis();
 
     Map<String, AvscParseResult> allNamedSchemas = context.getAllNamedSchemas();
+
+    // Handle duplicate schemas
     Map<String, List<AvscParseResult>> duplicates = context.getDuplicates();
+    for (Map.Entry<String, List<AvscParseResult>> duplicateEntry : duplicates.entrySet()) {
+      String fqcn = duplicateEntry.getKey();
+      StringBuilder allFilesString = new StringBuilder();
+      for (AvscParseResult avscParseResult : duplicateEntry.getValue()) {
+        allFilesString.append(avscParseResult.getURI());
+        allFilesString.append(", ");
+      }
+      if (!config.getDuplicateSchemasToIgnore().contains(fqcn)) {
+        switch (config.getDupBehaviour()) {
+          case FAIL:
+            throw new RuntimeException("ERROR: schema " + fqcn + " found in 2+ places: " + allFilesString);
+          case WARN:
+            System.err.println("WARNING: schema " + fqcn + " found in 2+ places: " + allFilesString);
+            break;
+          case FAIL_IF_DIFFERENT:
+            String baseSchema = null;
+            AvscParseResult baseSchemaResult = null;
+            for (AvscParseResult duplicateParseResult : duplicateEntry.getValue()) {
+              if (baseSchema == null) {
+                baseSchema = new AvscSchemaWriter().generateAvsc(duplicateParseResult.getTopLevelSchema(),
+                    AvscWriterConfig.CORRECT_MITIGATED);
+                baseSchemaResult = duplicateParseResult;
+                continue;
+              }
+
+              String currSchema = new AvscSchemaWriter().generateAvsc(duplicateParseResult.getTopLevelSchema(),
+                  AvscWriterConfig.CORRECT_MITIGATED);
+              if (!baseSchema.equals(currSchema)) {
```

Comment:
this compares top level schemas across files. the dup schema might be nested in any of them ?

Reply:
nit - i'd add a TODO to get back here when we have proper schema canonicalization, and compare canonical forms